modules/engage-serve-ui-config/src/main/java/org/opencastproject/engage/serveuiconfig/UIConfigRest.java

```diff
@@ -0,0 +1,165 @@
+
+  /** The currently used path to the configuration folder */
+  private String uiConfigFolder = "";
+
+  /** The used SecurityService */
+  private SecurityService securityService;
+
+  protected void setSecurityService(SecurityService securityService) {
+    this.securityService = securityService;
+  }
+
+
+  /**
+   * OSGI callback for activating this component
+   *
+   * @param cc
+   *          the osgi component context
+   */
+  public void activate(ComponentContext cc) throws ConfigurationException {
+    uiConfigFolder = cc.getBundleContext().getProperty(UI_CONFIG_FOLDER_PROPERTY);
+
+    if (StringUtils.isEmpty(uiConfigFolder)) {
+      String karafetc = cc.getBundleContext().getProperty("karaf.etc");
+
+      if (StringUtils.isBlank(karafetc)) {
+        throw new ConfigurationException("org.opencastproject.engage.serveuiconfig.uiconfigfolder not set and unable to"
+                                         + " fall back to default location based on ${karaf.etc}");
+      }
+
+      uiConfigFolder = new File(karafetc, UI_CONFIG_FOLDER_DEFAULT).getAbsolutePath();
+    }
+
+    logger.info("UI configuration folder is '{}'", uiConfigFolder);
+  }
+
+  @GET
+  @Path("{component}/{filename}")
+  @Produces(MediaType.WILDCARD)
+  @RestQuery(name = "getConfigFile",
+             description = "Returns the requested configuration file (json, css, etc..)",
+             pathParameters = {
+               @RestParameter(description = "Name of the component, which the configuration file belongs to",
+                              isRequired = true, name = "component", type = RestParameter.Type.STRING),
+               @RestParameter(description = "Name of the configuration file", isRequired = true,
+                              name = "filename", type = RestParameter.Type.STRING)
+             },
+             reponses = {
+               @RestResponse(description = "the requested configuration file", responseCode = HttpServletResponse.SC_OK),
+               @RestResponse(description = "if the configuration file doesn't exist", responseCode = HttpServletResponse.SC_NOT_FOUND),
+             },
+             returnDescription = "")
+  public Response getConfigFile(@PathParam("component") String component, @PathParam("filename") String filename) throws IOException, NotFoundException {
+    try {
+      String orgId = securityService.getOrganization().getId();
+      String basePath = new File(uiConfigFolder, orgId).getCanonicalPath();
+
+      File configFile = new File(PathSupport.concat(new String[] { uiConfigFolder, orgId, component, filename }));
+      String configFileCanPath = configFile.getCanonicalPath();
+
+      // is configFile a subdirectory of basePath (additional directory traversal protection), if not stop
+      if (!configFileCanPath.startsWith(basePath))
+      {
+        logger.warn("Directory traversal prevented");
+        throw new AccessDeniedException(configFileCanPath);
```

Comment:
Not sure about this, but I think this might be logged. Since this is a public endpoint that would make it  easy to spam the logs with rather large stack traces. Hence, if it is, just returning an access denied response may be better.

Reply:
this is just an additional directory traversal attack prevention, the PathParams should normally prevent directory traversal